src/components/Search/Search.tsx

```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+import {TeamsType, UserDataType} from 'types/types';
+
+interface SearchProps<ObjectType> {
+    object: ObjectType[];
+    updateObject: (object: ObjectType[]) => void;
+    notifyError: (hasError: boolean) => void;
+    searchProp: string;
+    placeholder: string;
+}
+
+const Search = <ObjectType extends TeamsType | UserDataType>(props: SearchProps<ObjectType>): JSX.Element => {
+    const {object, updateObject, notifyError, searchProp, placeholder} = props;
+
+    const [searchValue, setValue] = React.useState<string>('');
+
+    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => setValue(event.target.value);
+
+    React.useEffect(() => {
+        if (searchValue !== '') {
+            const filteredObject = object.filter((obj) => obj[searchProp].toLowerCase().includes(searchValue));
+            if (filteredObject.length === 0) {
+                notifyError(true);
```

Comment:
if no object is found, notify parent component about it, showing as an error from the search